provider/source/src/cldr_serde/currencies/supplemental.rs

```diff
@@ -1,56 +0,0 @@
-// This file is part of ICU4X. For terms of use, please see the file
-// called LICENSE at the top level of the ICU4X source tree
-// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
-
-//! Serde structs representing CLDR JSON numbers.json files.
-//!
-//! Sample file:
-//! <https://github.com/unicode-cldr/cldr-core/blob/master/supplemental/currencyData.json>
-
-use serde::Deserialize;
-use std::collections::BTreeMap;
```

Comment:
Issue: if we're not using these, it's a bug. Ideally the datagen code would have been added in the same PR as the safe serde code, but that doesn't mean we should just delete the impls. Please at least make them cfg test with a little test?

Reply:
I have no idea what these are, I'm certainly not going to start writing tests for them.

One of these files is not even included as module anywhere.

Reply:
They are for parsing supplemental currency data. I've been asking for @younies to finish up the currency component and he probably added these impls but doesn't use them yet, but we will be needing them.

Reply:
We should slap `allow(dead_code)` on this file, link to the currency issue, and move on